examples/pre-commit.nix

```diff
@@ -1,15 +0,0 @@
-{ inputs, config, ... }: {
```

Comment:
I moved all of this to `./pre-commit.nix` to keep all code quality things in one place